File: api-server/src/config/ecs.ts
export const ecsConfig = {
    cluster: process.env.ECS_CLUSTER_NAME ?? "",
    taskDefinition: process.env.ECS_TASK_DEFINITION ?? "",
    imageName: process.env.ECS_IMAGE_NAME ?? "",
    subnets: process.env.ECS_SUBNETS?.split(",").map((subnet) => subnet.trim()) ?? [],
    securityGroups: process.env.ECS_SECURITY_GROUPS?.split(",").map((sg) => sg.trim()) ?? [],
    launchType: process.env.ECS_LAUNCH_TYPE ?? "FARGATE",
    count: parseInt(process.env.ECS_COUNT ?? "1"),
    assignPublicIp: process.env.ECS_ASSIGN_PUBLIC_IP ?? "ENABLED",
}

File: api-server-go/internal/config/ecs.go
package config

import (
	"os"
	"strconv"
)

type ECSConfig struct {
	Cluster        string
	TaskDefinition string
	Subnets        string
	SecurityGroup  string
	AssignPublicIP string
	ImageName      string
	LaunchType     string
	Count          int
}

func getEnvOrDefault(key, defaultValue string) string {
	if value := os.Getenv(key); value != "" {
		return value
	}
	return defaultValue
}

func getEnvAsInt(key string, defaultValue int) int {
	if value := os.Getenv(key); value != "" {
		if intVal, err := strconv.Atoi(value); err == nil {
			return intVal
		}
	}
	return defaultValue
}

func GetECSConfig() ECSConfig {
	return ECSConfig{
		Cluster:        getEnvOrDefault("ECS_CLUSTER_NAME", ""),
		TaskDefinition: getEnvOrDefault("ECS_TASK_DEFINITION", ""),
		Subnets:        getEnvOrDefault("ECS_SUBNETS", ""),
		SecurityGroup:  getEnvOrDefault("ECS_SECURITY_GROUPS", ""),
		AssignPublicIP: getEnvOrDefault("ECS_ASSIGN_PUBLIC_IP", "ENABLED"),
		ImageName:      getEnvOrDefault("ECS_IMAGE_NAME", ""),
		LaunchType:     getEnvOrDefault("ECS_LAUNCH_TYPE", "FARGATE"),
		Count:          getEnvAsInt("ECS_COUNT", 1),
	}
}
